include/modem/location.h

```diff
@@ -494,20 +494,40 @@ struct location_config {
  */
 typedef void (*location_event_handler_t)(const struct location_event_data *event_data);
 
+/**
+ * Register handler for location events.
+ *
+ * This function is not needed for clients that call location_init() and pass event handler into it.
+ *
+ * @param[in] handler Event handler.
+ */
+int location_register_handler(location_event_handler_t handler);
+
+/**
+ * De-register handler for location events.
+ *
+ * @param[in] handler Event handler.
+ *
+ * @retval 0 if successful.
+ * @retval -ENXIO if the handler was not found.
+ * @retval -EINVAL if the handler was a @c NULL pointer.
+ */
+int location_deregister_handler(location_event_handler_t handler);
+
 /**
  * @brief Initializes the library.
  *
  * @details Initializes the library and sets the event handler function.
  * The first call to this function must be called before going to LTE normal mode.
  * This can be called multiple times, which sets the event handler again.
  *
- * @param[in] event_handler Event handler function.
+ * @param[in] handler Event handler function.
  *
  * @return 0 on success, or negative error code on failure.
  * @retval -EINVAL Given event_handler is NULL.
  * @retval -EFAULT Failed to obtain Wi-Fi interface.
  */
-int location_init(location_event_handler_t event_handler);
+int location_init(location_event_handler_t handler);
```

Comment:
If you decide to remove the `handler` parameter, it should be documented in `migration_guide_2.6.rst` what existing users need to do

Reply:
I wanted to remove this but didn't because I've been moralizing in other PRs where people break backwards compatibility that we do it too lightly :)
Then, I wanted to do proper deprecation and removal after two releases but I don't think that's an option because we would need to make another init function with a different name and that would live forever.
@jtguggedal, would you remove the parameter?

Reply:
I was thinking along the same lines as you. Removing the parameter has significant overhead if we follow best practice, like deprecating this function, introducing a new one, migration guide etc.
So I would probably just keep it

Reply:
I guess you are right. It still hurts my eyes. 😄

Reply:
It hurts me just as much if that helps 😄